File: app/src/main/java/com/myapplication/viewModel/LoginViewModel.java
package com.myapplication.viewModel;

import android.app.Activity;
import android.content.Context;
import android.databinding.ObservableBoolean;
import android.databinding.ObservableField;
import android.databinding.ObservableInt;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.View;

import com.myapplication.BuildConfig;
import com.myapplication.LoginActivity;
import com.myapplication.R;
import com.myapplication.adapter.ImageListAdapter;
import com.myapplication.adapter.ListAdapter;
import com.myapplication.bean.LoginBean;
import com.myapplication.command.LoginCommand;
import com.myapplication.http.RetrofitUtils;
import com.myapplication.utils.NdkUtils;

import java.util.ArrayList;
import java.util.List;

import io.reactivex.Observable;
import io.reactivex.Observer;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

/**
 * Created by Adminidtrator on 2017/10/23.
 */

public class LoginViewModel {
    private Context activity;

    LoginBean loginBean;
    //model
//    private LoginBean bean;

    /**
     * viewstyle
     * 用来控制view的显示样式
     */
    public final ViewStyle viewStyle = new ViewStyle();

    public void destoryAdapter() {
        adapter.set(null);
    }

    public class ViewStyle{
        public final ObservableBoolean isRefreshing = new ObservableBoolean();
        public final ObservableBoolean progressRefreshing = new ObservableBoolean();
        public final ObservableInt showListView = new ObservableInt();
        public final ObservableInt backgroundClolr= new ObservableInt();
    }
    //数据
    public final ObservableField<String> ImageUrl =new ObservableField<>();
    public final ObservableField<String> title = new ObservableField<>();

    public final ObservableField<LoginBean> bean = new ObservableField<LoginBean>() ;
    public final ObservableField<ListAdapter> adapter = new ObservableField<>();

    public final ObservableField<RecyclerView.LayoutManager> manager = new ObservableField<>();
    public final ObservableField<ImageListAdapter> iamgeAdapter= new ObservableField<>();


    //路由事件
    public final LoginCommand command = new LoginCommand(this);


    public LoginViewModel(Context activity, long id) {
        this.activity = activity;
        loadData(id);
    }

    private void loadData(long id) {
        loginBean = new LoginBean();
        loginBean.setToken(NdkUtils.daozhuanString());
        viewStyle.showListView.set(View.GONE);
        manager.set(new LinearLayoutManager(activity));
        iamgeAdapter.set(new ImageListAdapter(activity));
        viewStyle.backgroundClolr.set(activity.getColor(R.color.colorPrimary));
        if(BuildConfig.DEBUG){
            loginBean.setToken("debug");
        }else {
            loginBean.setToken("release");
        }
        List<LoginBean> list= new ArrayList<>();
        for(int i =0 ;i<5;i++){
            list.add(loginBean);
        }
        title.set("title");
        adapter.set(new ListAdapter(activity,list));
        bean.set(loginBean);
    }
    public void changeString(String tpUser){
        bean.get().setToken(tpUser);
    }
    public void updateUserMeddage(){
       Call call= RetrofitUtils.getInstance().getHelp().getLoginString("15936562980","333333");
       call.enqueue(new Callback<LoginBean>() {
           @Override
           public void onResponse(Call<LoginBean> call, Response<LoginBean> response) {
               loginBean=response.body();
               Log.d("TAG",""+loginBean.getToken());

           }

           @Override
           public void onFailure(Call<LoginBean> call, Throwable t) {

           }
       });
    }


    public void onDestory() {
        adapter.get().destory();
        activity = null;
    }

}

File: app/src/main/java/com/myapplication/viewModel/LoginViewModel.kt
package com.myapplication.viewModel

import android.app.Activity
import android.content.Context
import android.databinding.ObservableBoolean
import android.databinding.ObservableField
import android.databinding.ObservableInt
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.View
import android.widget.Toast

import com.myapplication.BuildConfig
import com.myapplication.LoginActivity
import com.myapplication.R
import com.myapplication.adapter.ImageListAdapter
import com.myapplication.adapter.ListAdapter
import com.myapplication.bean.LoginBean
import com.myapplication.command.LoginCommand
import com.myapplication.http.RetrofitUtils
import com.myapplication.utils.MylayoutManager
import com.myapplication.utils.NdkUtils
import com.myapplication.utils.NdkUtilsForKotlin

import io.reactivex.Observable
import io.reactivex.Observer
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.util.*

/**
 * Created by Adminidtrator on 2017/10/23.
 */

class LoginViewModel(var activity: Context?, id: Long) {

    internal var loginBean: LoginBean? = null
    //model
    //    private LoginBean bean;

    /**
     * viewstyle
     * 用来控制view的显示样式
     */
    val viewStyle = ViewStyle()
    //数据
    val ImageUrl = ObservableField<String>()
    val title = ObservableField<String>()

    val bean = ObservableField<LoginBean>()
    val adapter = ObservableField<ListAdapter>()

    val manager = ObservableField<MylayoutManager>()
    val iamgeAdapter = ObservableField<ImageListAdapter>()


    //路由事件
    val command = LoginCommand(this)

    fun destoryAdapter() {
        adapter.set(null)
    }

    inner class ViewStyle {
        val isRefreshing = ObservableBoolean()
        val progressRefreshing = ObservableBoolean()
        val showListView = ObservableInt()
        val backgroundClolr = ObservableInt()
    }


    init {
        loadData(id)
        checkVersion()
    }

    private fun checkVersion() {
    var  call = RetrofitUtils.getInstance().help.getLoginString("15936562980","333333")
        call.enqueue(object :Callback<LoginBean>{
            override fun onResponse(call: Call<LoginBean>?, response: Response<LoginBean>?) {
               if(null !=loginBean?.token){
                   Toast.makeText(activity,"要更新应用了", Toast.LENGTH_LONG).show()
               }
            }

            override fun onFailure(call: Call<LoginBean>?, t: Throwable?) {
               Toast.makeText(activity,t.toString(),Toast.LENGTH_SHORT)
            }

        })
    }

    private fun loadData(id: Long) {
        loginBean = LoginBean()
        loginBean!!.token = NdkUtils.helloWorld()
        viewStyle.showListView.set(View.GONE)
        manager.set(MylayoutManager())
//        manager.set(LinearLayoutManager(activity))
//        iamgeAdapter.set(ImageListAdapter(activity))
        viewStyle.backgroundClolr.set(activity!!.getColor(R.color.colorPrimary))
        if (BuildConfig.DEBUG) {
            loginBean!!.token = "debug"
        } else {
            loginBean!!.token = "release"
        }
        var list = ArrayList<LoginBean>()
        for (i in 0..4) {
            list.add(loginBean!!)
        }
        for(j  in 15 downTo 10){
            Log.d("Kotlin",""+j)
        }
        title.set("title")
        adapter.set(ListAdapter(activity, list))
        bean.set(loginBean)
    }

    fun changeString(tpUser: String) {
        bean.get().token = tpUser
    }

    fun updateUserMeddage() {
        val call = RetrofitUtils.getInstance().help.getLoginString("15936562980", "333333")
        call.enqueue(object : Callback<LoginBean> {
            override fun onResponse(call: Call<LoginBean>, response: Response<LoginBean>) {
                loginBean = response.body()
                Log.d("TAG", "" + loginBean!!.token)

            }

            override fun onFailure(call: Call<LoginBean>, t: Throwable) {

            }
        })
    }


    fun onDestory() {
        adapter.get().destory()
        activity = null
    }

}
